src/common/io.cc

```diff
@@ -251,7 +251,7 @@ int write_bytes(int fd, uint8_t *cursor, size_t length) {
   return 0;
 }
 
-int write_message(int fd, int64_t type, int64_t length, uint8_t *bytes) {
+int _write_message(int fd, int64_t type, int64_t length, uint8_t *bytes) {
```

Comment:
function names starting with an _ are reserved for the system/compiler, so we should name this something else (for example do_write_message)